pkg/infra/grn/grn_test.go

```diff
@@ -0,0 +1,72 @@
+package grn
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestParseGRNStr(t *testing.T) {
+	tests := []struct {
+		input     string
```

Comment:
pro tip! give the test cases a name so you can print it if the assertion fails :)